ci/cron/src/Main.hs

```diff
@@ -159,21 +161,28 @@ build_docs_folder path versions latest = do
                 then do
                     putStrLn "  Checks, reusing existing."
                     copy (old </> version) $ new </> version
-                    return $ Just gh_version
+                    return $ Just (gh_version, False)
                 else do
                     putStrLn "  Check failed. Rebuilding..."
-                    build version new
-                    return $ Just gh_version
+                    build version new latest_release_notes_sha
+                    return $ Just (gh_version, True)
             else do
                 putStrLn "  Not found. Building..."
-                build version new
-                return $ Just gh_version)
-        putStrLn $ "Copying latest (" <> latest <> ") to top-level..."
-        copy (new </> latest </> "*") (new <> "/")
+                build version new latest_release_notes_sha
+                return $ Just (gh_version, True))
+        putStrLn $ "Copying current (" <> current <> ") to top-level..."
+        copy (new </> current </> "*") (new <> "/")
         putStrLn "Creating versions.json..."
-        let (releases, snapshots) = List.partition (not . prerelease) documented_versions
-        create_versions_json releases (new </> "versions.json")
-        create_versions_json snapshots (new </> "snapshots.json")
+        let (releases, snapshots) = List.partition (not . prerelease . fst) documented_versions
+        create_versions_json (map fst releases) (new </> "versions.json")
+        create_versions_json (map fst snapshots) (new </> "snapshots.json")
+        let newly_built = fst $ head $ filter snd documented_versions
```

Comment:
Do we check before somewhere that at least one version is being rebuilt? If not this could crash.

Reply:
Unless I've made a horrible mistake somewhere, we should only enter this function if there is at least one release to build, so I decided otherwise crashing here would be the right thing to do.

I guess technically this could run if we have removed a version, and then it would crash and not update the docs site to remove it. I'll have to think a bit more for a proper fix. I thought I could use this as a shortcut, but I guess I won't be able to avoid searching the git history for the latest release.

Reply:
Happy to keep it this way, maybe just add a comment why the `head` is safe.

Reply:
Added a check.